src/NUnit.Xml.TestLogger/TestCaseNameParser.cs

```diff
@@ -123,7 +129,12 @@ public static ParsedName Parse(string fullyQualifedName)
                     }
                     else if (state == NameParseState.Parenthesis)
                     {
-                        if (thisChar == ')' || thisChar == '\\')
+                        if (thisChar == ')')
+                        {
+                            parenthesisCount++;
+                        }
+
+                        if (thisChar == '\\')
```

Comment:
With 70f258f, this condition may not be necessary. I believe build should be fixed too. Can we pull the latest master into this PR please?

Reply:
I have rebased this PR branch on the actual master and remove that condition. But two tests now failed, so I add one more commit to cover failed cases in Parse_FailsGracefullyOnNonParsableInputs_WithConsoleOutput

Reply:
And now CI builds finished successfully